cmd/mount.go

```diff
@@ -512,10 +512,18 @@ var mountCmd = &cobra.Command{
 				}()
 			}
 
+		retry:
+			// If the .pid file is locked and there no blobfuse process owning it then we need to try
+			// a cleanup of the .pid file. If cleanup goes through then retry the daemonization.
 			child, err := dmnCtx.Reborn()
 			if err != nil {
-				log.Err("mount : failed to daemonize application [%v]", err)
-				return Destroy(fmt.Sprintf("failed to daemonize application [%s]", err.Error()))
+				log.Err("mount : failed to daemonize application [%s], trying auto cleanup", err.Error())
+				rmErr := os.Remove(pidFileName)
+				if rmErr != nil {
+					log.Err("mount : auto cleanup failed [%v]", rmErr.Error())
+					return Destroy(fmt.Sprintf("failed to daemonize application [%s]", err.Error()))
+				}
+				goto retry
 			}
```

Comment:
Retry should be done only twice or fixed number of times.

Reply:
There is no retry here. Only if remove of file succeeds then we retry to daemonize which will succeed. If remove of pid file fails then we just error out. Ideally it shall never retry more than once.